lib/chef/resource/yum_package.rb

```diff
@@ -62,6 +62,21 @@ class YumPackage < Chef::Resource::Package
       property :allow_downgrade, [ true, false ], default: false
 
       property :yum_binary, String
+
+      private
+
+      def get_package_name(x)
+        if is_valid_package?(x)
+          Chef::Log.error("Invalid package name #{x}")
+          exit 0
```

Comment:
the exit 0 here is wrong.   this should probably be expressed as a `callback` validation argument instead of being in the `coerce` method so that it throws a proper validationfailed exception type.